### PR TITLE
DEPR: Remove xlwt

### PR DESCRIPTION
**Pull request by mroeschke**
Introduced in https://github.com/pandas-dev/pandas/pull/38317

### CODE REVIEW

File: pandas/tests/io/excel/test_writers.py [hunk 1]
```diff
@@ -61,7 +60,6 @@ def set_engine(engine, ext):
     [
         pytest.param(".xlsx", marks=[td.skip_if_no("openpyxl"), td.skip_if_no("xlrd")]),
         pytest.param(".xlsm", marks=[td.skip_if_no("openpyxl"), td.skip_if_no("xlrd")]),
-        pytest.param(".xls", marks=[td.skip_if_no("xlwt"), td.skip_if_no("xlrd")]),
```
Comment (WillAyd) [inline pandas/tests/io/excel/test_writers.py:64]: Guessing xlrd will be removed too but not part of this PR?

Comment (mroeschke) [reply pandas/tests/io/excel/test_writers.py:64]: I _think_ based on https://github.com/pandas-dev/pandas/pull/38317 that xlrd will continued to be supported since it's just for reading (although the package isn't maintained).

Comment (rhshadrach) [reply pandas/tests/io/excel/test_writers.py:64]: I'm good with keeping xlrd. However, I'm thinking we should warn users on any use that it is not maintained and bump the minimum version to 2.0 (so it can only be used to read xls files).

File: ci/code_checks.sh [hunk 2]
```diff
@@ -47,7 +47,7 @@ import pandas
 
 blocklist = {'bs4', 'gcsfs', 'html5lib', 'http', 'ipython', 'jinja2', 'hypothesis',
              'lxml', 'matplotlib', 'openpyxl', 'py', 'pytest', 's3fs', 'scipy',
-             'tables', 'urllib.request', 'xlrd', 'xlsxwriter', 'xlwt'}
```
Comment (jbrockmendel) [inline ci/code_checks.sh:50]: keeping xlrd here but removing it in the asv?

**Comment by jbrockmendel** [comment]
small question about xlrd in the asv, otherwise lgtm

File: ci/code_checks.sh (hunk 2 above)
Comment (mroeschke) [reply ci/code_checks.sh:50]: The asvs use `to_excel/read_excel` to roundtrip test `xlwt/xlrd`. Since `xlwt` is removed, the roundtrip is broken and I thought It's not worth adding a separate, large excel file to test xlrd.

File: pandas/tests/io/excel/test_writers.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/io/excel/test_writers.py:64]: That sounds good to me. Probably good to warn and bump of the min version in a separate PR.

**Review by WillAyd — APPROVED** [review]
lgtm
